### ISSUE TITLE
Issues with changing innerHTML

### ISSUE THREAD

**Issue by fstanis** [ampproject/worker-dom#621]
```html
<amp-script layout="container" script="myscript">
  <button id="btn">Press me</button>
</amp-script>

<script id="myscript" type="text/plain" target="amp-script">
  document.querySelector('#btn').addEventListener('click', () => {
    document.body.innerHTML += '<p>Hello</p>';
  });
</script>
```

Two things happen when the button is pressed:

1.  The button gets removed.
2.  `<q></q>` gets added as the first element of the body.

**Comment by morsssss** [comment]
Did anyone ever look into this? @kristoferbaxter ?

**Comment by cloewen8** [comment]
I don't believe `innerHTML` is (fully) supported. The web_compat_table shows it as "crossed out".

I really hope I'm wrong! Maybe the table is outdated, that is mislabeled or I'm misunderstanding what "crossed out" means (can't find a legend to be sure).

Having tested it, I also get the q element. But the text gets added, regardless of if I set it (`=`) or add it (`+=`).
![image](https://user-images.githubusercontent.com/1569808/90689666-a3ef5c00-e22d-11ea-9dbe-36912f73b9e0.png)

**Comment by morsssss** [comment]
I've been wondering about this too recently, as I've used `innerHTML` on many an occasion. But I've never used it on `document.body`.

With this HTML, though:

```html
  <amp-script layout="responsive" script="myscript" data-ampdevmode>
      <p><b><span>This text</span></b> will change.</p>
      <button>Change things</button>
  </amp-script>
```

This works as expected:

```js
span.innerHTML = "Some other random text";
```

as does this:

```js
span.innerHTML += "Some other random text<p>I am a paragraph</p>";
```

Is `innerHTML` fully supported? Or just mostly? In either case we should update the table.

/cc @samouri in case he has a chance to look at this...

**Comment by samouri** [comment]
`innerHTML` is partially supported. A good action item for us is to document all of the bits that aren't. 
E.g. I know decoding entities still isnt.

**Comment by morsssss** [comment]
For now, I've submitted https://github.com/ampproject/worker-dom/pull/911 ... as a start!
